crates/sql-sink/src/main.rs

```diff
@@ -1,24 +1,58 @@
+mod backoff;
 mod bind;
 mod config;
 mod db;
 mod insert;
 mod sink;
 mod upsert;
 
+use std::time::Duration;
+
+use anyhow::Result;
 use config::SqlConfig;
-use fluvio_connector_common::{connector, consumer::ConsumerStream, tracing::trace, Result, Sink};
-use fluvio_model_sql::Operation;
 use futures::SinkExt;
+
+use fluvio_connector_common::{
+    connector,
+    consumer::ConsumerStream,
+    tracing::{error, trace, warn},
+    Sink,
+};
+use fluvio_model_sql::Operation;
+
 use sink::SqlSink;
 
+use crate::backoff::Backoff;
+
+const BACKOFF_LIMIT: Duration = Duration::from_secs(1000);
+
 #[connector(sink)]
 async fn start(config: SqlConfig, mut stream: impl ConsumerStream) -> Result<()> {
-    let sink = SqlSink::new(&config)?;
-    let mut sink = sink.connect(None).await?;
-    while let Some(item) = stream.next().await {
-        let operation: Operation = serde_json::from_slice(item?.as_ref())?;
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = Backoff::new();
+
+    loop {
+        let wait = backoff.next();
+
+        if wait > BACKOFF_LIMIT {
+            error!("Max retry reached, exiting");
+        }
```

Comment:
Judging by the error message, there is a missing `return` here.

Reply:
Adjusted the backoff and error message so the connector will continue retrying - though we may need a long term company discussion on where we we want to default.

 I also added a reset of t he backoff time to a short timeframe if a successful connect occurs